LanguageExt.Core/DataTypes/List/Lst.Internal.cs

```diff
@@ -219,8 +219,11 @@ public LstInternal<A> InsertRange(int index, IEnumerable<A> items, Pred<A> pred)
         /// Remove an item from the list
         /// </summary>
         [Pure]
-        public LstInternal<A> Remove(A value) => 
-            Remove(value, Comparer<A>.Default);
+        public LstInternal<A> Remove(A value, IEqualityComparer<A> equalityComparer = null)
+        {
+            var i = IndexOf( value, equalityComparer: equalityComparer );
+            return i >= 0 ? RemoveAt( i )  : this;
```

Comment:
The problem with the new implementation is it traverses the tree twice.  I think we can do better than this.  Leave it with me.